### DISCUSSION TITLE
Can I create a CommandTree with commands.Bot

### DISCUSSION THREAD

**Discussion by rexwu1104** [Rapptz/discord.py#7785] [Q&A] [answered]
I create a CommandTree with an error, It says "This client already has an associated command tree."
And I use the AutoShardedBot class, but if I replace to AutoShardedClient, it doesn't give this error.

here is my code
```py
import asyncio
import discord_together
from discord import AutoShardedClient, Intents
from discord.ext import commands
from discord.app_commands import CommandTree
from config import *

# bot = AutoShardedClient()
bot = commands.AutoShardedBot(command_prefix="t!", intents=Intents.all())
# bot._connection._command_tree = None
Applications = CommandTree(bot)

@bot.event
async def on_ready():
    bot.togetherControl = await discord_together.DiscordTogether(token)
    print('new version!!!')

async def module_init():
    await bot.load_extension('Commands')
    
asyncio.run(module_init())
bot.run(token)
```

I want to  know why this situation ocurred, I look at the source code, but I still don't know.

If I execute line 9, it doesn't have any error.

**Comment by LostLuma** [comment, marked as answer]
As the error tells you, the `Bot` and `AutoShardedBot` classes already have a comand tree registered by default which is used to register application commands found in Cogs.

You can access it under [`Bot.tree`](https://discordpy.readthedocs.io/en/master/ext/commands/api.html#discord.ext.commands.Bot.tree) to add your commands, just as you would with a tree that you create yourself.

```py
bot = commands.AutoShardedBot(command_prefix="t!", intents=Intents.all())

@bot.tree.command()
async def example(interaction):
    ...
```

**Comment by joannx** [comment]
No need to do Applications = CommandTree(bot), theres already an existing attribute ``bot.tree``, if you want it to remain the same Applications definition, you can do ``Applications = bot.tree``

**Reply by rexwu1104** [reply]
Oh, I see, thanks.

**Comment by rexwu1104** [comment]
thank you all.
